batch/batch/front_end/front_end.py

```diff
@@ -1119,20 +1138,29 @@ async def parse_error(msg):
         return await parse_error(f'Invalid search; start must be earlier than end.')
 
     sql = f'''
-SELECT billing_project, user, CAST(COALESCE(SUM(msec_mcpu), 0) AS SIGNED) AS msec_mcpu
+SELECT
+  billing_project,
+  `user`,
+  CAST(COALESCE(SUM(IF(format_version < 3, msec_mcpu, 0)), 0) AS SIGNED) as msec_mcpu,
+  SUM(IF(format_version >= 3, `usage` * rate, NULL)) as cost
 FROM batches
-WHERE batches.`time_created` >= %s AND batches.`time_created` <= %s
-GROUP by billing_project, user;
+LEFT JOIN aggregated_batch_resources
+       ON aggregated_batch_resources.batch_id = batches.id
+LEFT JOIN resources
+        ON resources.resource = aggregated_batch_resources.resource
+GROUP BY billing_project, `user`;
 '''
+
     sql_args = (start, end)
 
-    def billing_record_to_dict(app, record):
-        msec_mcpu = record['msec_mcpu']
-        record['cost'] = cost_from_msec_mcpu(app, msec_mcpu)
+    def billing_record_to_dict(record):
+        cost_msec_mcpu = cost_from_msec_mcpu(record['msec_mcpu'])
+        cost_resources = record['cost']
+        record['cost'] = coalesce(cost_msec_mcpu, 0) + coalesce(cost_resources, 0)
```

Comment:
ok tiny thing, we have two coalesces here but one in the SQL, seems simpler to do it in one or the other. In particular, `cost_msec_mcpu` should always be a number, right? Maybe just remove the coalesce from SQL? `CAST(NULL AS SIGNED)` is still `NULL`, so I think it shouldn't affect the semantics.